src/prerender.ts

```diff
@@ -201,7 +201,10 @@ export async function prerender(nitro: Nitro) {
 
     const filePath = join(nitro.options.output.publicDir, _route.fileName);
     await writeFile(filePath, Buffer.from(_route.data));
-    nitro._prerenderedRoutes.push(_route);
+    nitro._prerenderedRoutes.push({
+      route: _route.route,
+      fileName: _route.fileName,
```

Comment:
We could add synchronous getters that reads from the fs to access the properties we are removing, to avoid breaking integrations that rely on this API.

Reply:
Nice idea. Please see my comments in https://github.com/unjs/nitro/pull/1536, it makes sure data and contents are available until first read. I would be more than happy to add compatibility sync read if an integration needs contents after the extend hook (and write). Didn't add it initially to avoid unwanted performance overhead of sync operation (specially that it is a hidden getter and to avoid memory leaks, we cannot cache return value. it should be sync read each time accessing getter which is costly (and hidden!)

Reply:
Not sure whether to comment here or there, but I think it is indeed worth adding a getter (and a console log deprecating the properties or warning about hidden host of access?) until a future version as otherwise it is a breaking behaviour change.

Reply:
(feel free to open an issue to not forget), checking you are right `prerender:route` will be broken but i guess we can fix it by reordering where the hook is being called to avoid adding that workaround. This is only hook that is currently being called before disk write.

Reply:
https://github.com/unjs/nitro/pull/1537 should fix it for that hook.